pkg/prometheus/promcfg.go

```diff
@@ -2150,12 +2150,16 @@ func (cg *ConfigGenerator) generateAlertmanagerConfig(alerting *monitoringv1.Ale
 		cfg = cg.WithMinimumVersion("2.48.0").addSigv4ToYaml(cfg, fmt.Sprintf("alertmanager/auth/%d", i), store, am.Sigv4)
 
 		apiVersionCg := cg.WithMinimumVersion("2.11.0")
-		switch am.APIVersion {
-		case "v1":
+		if am.APIVersion != nil {
+			switch monitoringv1.AlertmanagerAPIVersion(strings.ToUpper(string(*am.APIVersion))) {
 			// API v1 isn't supported anymore by Prometheus v3.
-			cfg = apiVersionCg.WithMaximumVersion("2.999.0").AppendMapItem(cfg, "api_version", am.APIVersion)
-		case "v2":
-			cfg = apiVersionCg.AppendMapItem(cfg, "api_version", am.APIVersion)
+			case monitoringv1.AlertmanagerAPIVersion1:
+				if cg.version.Major <= 2 {
+					cfg = apiVersionCg.AppendMapItem(cfg, "api_version", strings.ToUpper(string(*am.APIVersion)))
+				}
+			case monitoringv1.AlertmanagerAPIVersion2:
+				cfg = apiVersionCg.AppendMapItem(cfg, "api_version", strings.ToUpper(string(*am.APIVersion)))
```

Comment:
ToLower

Reply:
Thanks @simonpasquier 
I got it. 
Change to ToLower